universal-login-react/src/ui/TopUp/Fiat/TopUpWithFiat.tsx

```diff
@@ -51,6 +52,7 @@ export const TopUpWithFiat = ({walletService, modalClassName, logoColor}: TopUpW
         </>
       );
     case TopUpProvider.RAMP:
+      setHeaderVisible(false);
```

Comment:
Side effects in render are bad practice

Reply:
Instead do `useEffect` on `modal` and call it there

Reply:
Also not sure if a callback to the parent component is the best solution here. Consider moving the state up and/or decomposing this component into two: `TopUpFiatDetails` and `TopUpWithOnRampProvider`

Reply:
Good point with `useEffect` - done